<?xml version="1.0"?>
<launch>
    <!-- Obstacle Lifespan Module (OLM) and its core components -->

    <!-- It changes a whole package Logger level  -->
    <!--<env name="ROSCONSOLE_CONFIG_FILE" value="$(find simulation)/config/custom_rosconsole.conf"/>-->

    <!-- MongoDb server-->
    <include file="$(find mongodb_store)/launch/mongodb_store.launch"/>

    <!-- Image recognition -->
    <node pkg="find_object_2d" type="find_object_2d" name="find_object_2d">
        <param name="gui" value="false"/>
        <param name="subscribe_depth" value="true"/>
        <param name="objects_path" value="$(find simulation)/image_rec/"/>

        <remap from="image" to="/camera/rgb/image_raw"/>
        <remap from="rgb/image_rect_color" to="/camera/rgb/image_raw"/>
        <remap from="depth_registered/image_raw" to="/camera/depth/image_raw"/>
        <remap from="depth_registered/camera_info" to="/camera/depth/camera_info"/>
    </node>

    <!-- Obstacle detection and its position calculation -->
    <node name="obstacle_dpc" pkg="simulation" type="obstacle_dpc_node" output="screen"/>

    <!-- Job scheduler -->
    <node name="job_scheduler" pkg="simulation" type="job_scheduler_node" output="screen"/>

    <!-- Job catcher -->
    <node name="job_catcher" pkg="simulation" type="job_catcher_node" output="screen"/>

</launch>